### PR TITLE
[reposync] Check GPG signatures of downloaded packages (RhBug:1856818) 

### PR DESCRIPTION
**Pull request by m-blaha**
YUMv3 reposync used to have --gpgcheck option to remove packages that fail GPG signature checking after downloading.
This patch implements the option for DNF.

https://bugzilla.redhat.com/show_bug.cgi?id=1856818

Tests: https://github.com/rpm-software-management/ci-dnf-stack/pull/900

### CODE REVIEW

File: doc/reposync.rst [hunk 1]
```diff
@@ -48,6 +48,9 @@ All general DNF options are accepted. Namely, the ``--repoid`` option can be use
 ``--download-metadata``
     Download all repository metadata. Downloaded copy is instantly usable as a repository, no need to run createrepo_c on it.
 
+``-g, --gpgcheck``
+    Remove packages that fail GPG signature checking after downloading. Exit code is ``1`` if at least one package was removed.
```
Comment (kontura) [inline doc/reposync.rst:52]: I suggest we should mention here that if a specific repo has `gpgcheck=0` its packages are not checked even with this option.

Or perhaps this option should override the `gpgcheck` config value and check even packages from repos with `gpgcheck=0`?

Comment (m-blaha) [reply doc/reposync.rst:52]: May be the cleanest solution would be to do the gpgcheck by default and skip it only when `--nogpgcheck` option was used. In that case `--gpgcheck` option would not be needed at all. But I do not want to change the reposync behaviour during RHEL 8 cycle.
So I will go with clarifying this in documentation.

**Comment by kontura** [comment]
@rh-atomic-bot r+

**Comment by kontura** [comment]
The tests pass locally, LGTM.

**Pull request merged**